src/Adapters/Auth/AuthAdapter.js

```diff
@@ -1,20 +1,114 @@
 /*eslint no-unused-vars: "off"*/
 export class AuthAdapter {
-  /*
+  constructor() {
+    /**
+     * If set to true the signup/login
+     * will fail if user do not provide
+     * authData associated to this Auth Adapter
+     */
+    this.required = false;
+
+    /**
+     * Set to true if you want to force to validateAuthData
+     * even if authData do not change
+     */
+    this.alwaysValidate = false;
+
+    /**
+     * Usage policy
+     * default: can be combined with ONE additional auth provider if additional configured on user
+     * additional: could be only used with a default policy auth provider
+     * solo: Will ignore ALL additional providers if additional configured on user
+     */
+    this.policy = 'default';
+  }
+  /**
   @param appIds: the specified app ids in the configuration
   @param authData: the client provided authData
   @param options: additional options
+  @param req: RestWrite instance with config/auth/data
   @returns a promise that resolves if the applicationId is valid
    */
-  validateAppId(appIds, authData, options) {
+  validateAppId(appIds, authData, options, req) {
     return Promise.resolve({});
   }
 
-  /*
+  /**
```

Comment:
@dblythy  does the new following validate functions seems more clear to you to implement MFA ?

Reply:
Not entirely yet, I haven't looked through it completely. I'll probably wait until this is merged and then create a PR for a MFA adapter based on the final interface and the test code that you've written for it. Is that ok?

Reply:
Yeah no problem !

Reply:
Thank you legend! 😊

Reply:
I might have another look 😊